src/parser/Tokenizer.cc

```diff
@@ -333,3 +333,60 @@ Parser::Tokenizer::udec64(const char *description, const SBuf::size_type limit)
     return result;
 }
 
+/// NextWordWhileRemovingDoubleQuotesAndBackslashesInsideThem() helper that
+/// parses an optional "quoted string" sequence at the start of its input.
+static std::optional<SBuf>
+UnquoteSequence_(Parser::Tokenizer &tk)
+{
+    if (!tk.skipOne(CharacterSet::DQUOTE))
+        return std::nullopt;
+
+    SBuf decoded;
+    while (!tk.atEnd()) {
+        SBuf prefix;
+        static const auto backslashChar = CharacterSet("backslash", "\\");
+        static const auto ordinaryChars = (CharacterSet::DQUOTE + backslashChar).complement();
+        if (tk.prefix(prefix, ordinaryChars))
+            decoded.append(prefix);
+
+        if (tk.skipOne(CharacterSet::DQUOTE))
+            return decoded; // well-formed "quoted string"
+
+        if (tk.skipOne(backslashChar)) {
+            SBuf escaped;
+            static const auto allChars = CharacterSet().complement("any char");
+            if (tk.prefix(escaped, allChars, 1))
+                decoded.append(escaped);
+            else
+                Assure(tk.atEnd()); // input ends at backslash (which we skip)
+        }
+    }
+
+    return decoded; // truncated "quoted string" (no closing quote); may be empty
+}
+
+std::optional<SBuf>
+NextWordWhileRemovingDoubleQuotesAndBackslashesInsideThem(Parser::Tokenizer &tk)
```

Comment:
I am using this very long name because I assume that the underlying syntax here does not have a well-known name (it is effectively defined by existing strwordtok() _implementation_ that this code is based on) and to automatically flag all callers for extra scrutiny: If the caller should use a well-known syntax instead, they should not call this function.

Just like with libcSpace(): We may (and probably should!) change callers grammar to some well-known alternative, but let's not do that in this behavior-preserving PR.

If my assumption about the lack of a well-known name for this syntax is wrong, please let me know what the correct name is, and I will rename this function in his PR, of course.

Reply:
Er, this is called "un-escaping a quoted string"

Reply:
> Er, this is called "un-escaping a quoted string"

Yes, it is. A lot of things can be (and are!) called that. This function name should be more precise IMO, for the reasons detailed earlier.